tests/masonry/builders/test_sdist.py

```diff
@@ -587,3 +587,19 @@ def test_sdist_disable_setup_py():
             "my-package-1.2.3/PKG-INFO",
             "my-package-1.2.3/my_package/__init__.py",
         }
+
+
+def test_split_source():
+    root = fixtures_dir / "split_source"
+    poetry = Factory().create_poetry(root)
+
+    builder = SdistBuilder(poetry)
+
+    # Check setup.py
+    setup = builder.build_setup()
+    setup_ast = ast.parse(setup)
+
+    setup_ast.body = [n for n in setup_ast.body if isinstance(n, ast.Assign)]
+    ns = {}
+    exec(compile(setup_ast, filename="setup.py", mode="exec"), ns)
+    assert "module_a" in ns["package_dir"] or "module_b" in ns["package_dir"]
```

Comment:
(Thanks for putting this together - I was having the same issue with two packages lying in different folders.)

Why is there an `or` here?  Shouldn't it always be that `ns["package_dir"].keys() == {"", "module_b"}`?  If not, that'd mean that the build is not reproducible given the same input, which isn't good.

Reply:
Yes, I believe you are correct because the first entry in the list is module_a which will always be added as "".

Reply:
Well, you could just test for equality: `ns["package_dir"] == {"": "lib_a", "module_b": "lib_b/module_b"}`.